### ISSUE TITLE
Support for bulk email personalizations?

### ISSUE THREAD

**Issue by travisp** [eddiezane/sendgrid-actionmailer#59]
One of the nice features of the sendgrid API is that it's possible to send multiple emails to different recipients using personalizations. For example, suppose you have a feature where a group of users want to be notified when something happens. Instead of looping through each user and making a separate API call for each one, personalizations can be used to make one API call that sends a separate email to each user (and substitutes in things like their first name).

Would there be interest in having this gem to support this?

I would do this by extending `mail` to take `is_multiple` similar to the sendgrid python library, e.g.
```
addresses = ['email1@example.com', 'email2@example.com']
sections = [{'%fname%' => 'User 1'}, {'%fname%' => 'User 2'}]
mail(to: addresses, sections: sections, subject: 'email subject', body: 'email body', is_multiple: true)
```

and modifying the `to_personalizations` method of sendgrid-actionmailer

**Comment by tyrauber** [comment]
Hi @travisp, pull requests are welcome!

I am wondering if something like this might be best accomplished with dynamic templates and dynamic_template_data?  As of now, dynamic template personalization works on a mail by mail basis, but I see no reason why an array couldn't be passed to 'dynamic_template_data' and have an email attribute override the 'to' attribute.

```
data = [{'email' =>'email1@example.com',  'fname' => 'User 1'}, {'email' => 'email2@example.com', 'fname' => 'User 2'}]
mail(subject: 'test', body: 'test',  template_id: 'XXXXXXXXX', dynamic_template_data: data)
```

**Comment by travisp** [comment]
@tyrauber I think that could work great for a lot of people. The challenge here is that I prefer not to use templates and to keep the email content in my source code. I don't know how common email templates usage is (it seems to depend on the size and makeup of the team that works on a project).

Without holding you to it, are you open to the `is_multiple` approach? I'd be happy to put together a PR.

**Comment by tyrauber** [comment]
Absolutely. Pull requests are appreciated! I think adding the ability to send multiple emails to different recipients using personalizations, would be a significant improvement. The only thing I question is implementation. 

Looking at the [personalization api examples](https://sendgrid.com/docs/for-developers/sending-email/personalizations/#sending-two-different-emails-to-two-different-groups-of-recipients), and [the sendgrid-ruby email example](https://github.com/sendgrid/sendgrid-ruby/blob/master/examples/mail/mail.rb#L94-L130), it looks like personalizations takes an array of hashes.


```
{
  "personalizations": [{
    "subject": "Hello, World!", 
    "substitutions": {
      "id": "substitutions", 
      "type": "object"
    }, 
    "to": [
      {
        "email": "john.doe@example.com", 
        "name": "John Doe"
      }
    ]
  }, ...]
}
```

I am wondering if we should embrace this methodology and pass the personalizations array to mail.

The benefit being that:

 - the implementation would mirror the sendgrid v3 api and official sendgrid-ruby gem
 - both transactional and template emails could then support the same interface for bulk sending
 - a is_multiple flag wouldn't be required
 - and a single array of hashes would limit the potential for bulk sending errors

Thoughts?

**Comment by travisp** [comment]
I'm happy to take that approach. Although I think there's something simple with `is_multiple`, I agree that it would have more potential for confusion or error. I'll see if I can get up a PR for feedback.

**Comment by tyrauber** [comment]
Merged in #60. Minor version bump to 2.5.0 pushed.
